docs/proposal/20210823-gpu-support.md

```diff
@@ -0,0 +1,164 @@
+# GPU support CAPV
+
+```text
+---
+title: GPU support CAPV
+authors:
+  - "@geetikabatra"
+reviewers:
+  - "@vijaykumar"
+  - "@ankitaswamy"
+  - "@ashutoshkumar"
+  
+creation-date: 2021-08-23
+last-updated: 2021-08-25
+status: WIP
+```
+
+## Table of Contents
+
+* [Single Controller Multitenancy](#gpu-support)
+  * [Glossary](#glossary)
+  * [Summary](#summary)
+  * [Motivation](#motivation)
+    * [Goals](#goals)
+  * [Proposal](#proposal)
+    * [User Stories](#user-stories)
+      * [Story 1 - using vGPU](#story-1---using-vgpu)
+      * [Story 2 - GPU Direct implementation](#story-2---gpu-direct-implementation)
+      * [Story 3 - PCI passthrough for single node Customer](#story-3---pci-passthrough-for-single-node-customer)
+    * [Requirements](#requirements)
+      * [Functional Requirements](#functional-requirements)
+      * [Non-Functional Requirements](#non-functional-requirements)
+    * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+      * [Controller Changes](#controller-changes)
+  * [Additional Details](#additional-details)
+    * [Test Plan](#test-plan)
+    * [Graduation Criteria](#graduation-criteria)
+  * [Implementation History](#implementation-history)
+
+## Glossary
+
+* CAPV - An abbreviation of Cluster API Provider vSphere
+
+## Summary
+
+CAPV currently does not support GPU-based infrastructure.
+
+This proposal outlines ways to provision GPU support in CAPV. The proposed changes will maintain backward compatibility and maintain the existing behavior without any extra user configuration.
+
+## Motivation
+
+CAPV does not currently support GPU-accelerated workloads which have a limited addressable market. Competitively, all of the hyper-scale cloud providers (AWS, Azure, & GCP) offer GPU-accelerated virtualization and Kubernetes platforms. Many of them additionally offer robust vertically integrated AI/ML solutions building on open source technology. GPU support is a long-requested feature. With this capability, users can leverage GPU support.
+
+### Goals
+
+1. To enable GPU Support on Vsphere Kubernetes Cluster.
+2. To allow vGPU support for multi-node clusters.
+3. To allow GPU Direct support for multi-node clusters.
+4. To allow PCI passthrough support for single-node clusters.
+
+## Proposal
+
+### User Stories
+
+#### Story 1 - using vGPU
+
+Stacy works at an IT department of an organization where AI/ML models are run every hour and the entire staff wants to leverage GPU-supported nodes. VGPU's are able to provide close to bare-metal performance. Stacy needs a lot of sharing GPUs.
+vGPU provides a full-fleshed cluster from which a virtual GPU can be requested.
```

Comment:
Requesting a vGPU does not necessarily mean that it can be shared across, right? 
One can still request a vGPU and it may not be shareable.  
For example, in the below doc ( specific to Nvidia ), I can see that we can request vGPU in multiple modes and one of the mode can allow vGPU to be shared. 
https://docs.nvidia.com/grid/latest/grid-vgpu-user-guide/index.html

The question I am trying to ask is that, when we say that we have a vGPU available, does it inherently mean that it is shareable? 
Are we trying to say that from CAPV end user prespective, when a user request for a vGPU that always means it is shareable ( we make sure from out implementation that we provision vGPU in a shareable mode ), which sounds fine to me.

Reply:
Inherently it's not shareable. Sharing is enabled inside of physical hardware using one of the above techniques you pointed out, which is the user's own choice. We assume it to be just a piece of GPU(shareable or non-shareable).  Sharing should be automatically handled by the vGPU instance(Need to double-check that).